pages/api/v1/activity-feed/subscribed-posts.ts

```diff
@@ -0,0 +1,323 @@
+							data?.isSpam && !data?.isSpamReportInvalid ? Number(process.env.REPORTS_THRESHOLD || 50) : data?.isSpamReportInvalid ? 0 : data?.spam_users_count || 0,
+						status,
+						status_history: statusHistory,
+						summary: data?.summary || '',
+						tags: data?.tags || [],
+						tally,
+						timeline: proposalTimeline,
+						title: data?.title || subsquareTitle || null,
+						topic: topic
+							? topic
+							: isTopicIdValid(topic_id)
+							? {
+									id: topic_id,
+									name: getTopicNameFromTopicId(topic_id)
+							  }
+							: topicFromType,
+						track_no: !isNaN(trackNumber) ? trackNumber : null,
+						type: type || ProposalType.REFERENDUM_V2,
+						user_id: data?.user_id || 1
+					};
+
+					await getContentSummary(post, network, true);
+
+					if (!post?.content?.length) {
+						if (proposer) {
+							post.content = `This is a ${getProposalTypeTitle(
+								ProposalType.REFERENDUM_V2
+							)} whose proposer address (${proposer}) is shown in on-chain info below. Only this user can edit this description and the title. If you own this account, login and tell us more about your proposal.`;
+						} else {
+							post.content = `This is a ${getProposalTypeTitle(
+								ProposalType.REFERENDUM_V2
+							)}. Only the proposer can edit this description and the title. If you own this account, login and tell us more about your proposal.`;
+						}
+					}
+
+					if (!process.env.AI_SUMMARY_API_KEY) {
+						post.summary = post?.content;
+					}
+					post.content = '';
+
+					return post;
+				}
+			}
+			let subsquareTitle = '';
+			let subsquareContent = '';
+			const res = await getSubSquareContentAndTitle(ProposalType.REFERENDUM_V2, network, postId);
+			subsquareTitle = res?.title;
+			subsquareContent = res?.content;
+
+			const post: IActivityFeedPost = {
+				allowedCommentors: EAllowedCommentor.ALL,
+				assetId: assetId || null,
+				commentsCount: commentsQuerySnapshot.data()?.count || 0,
+				content: subsquareContent || '',
+				created_at: createdAt,
+				curator,
+				description: description || '',
+				end: end,
+				hash: hash || null,
+				higestSentiment: null,
+				identity,
+				method: subsquidPost?.preimage?.method,
+				parent_bounty_index: parentBountyIndex || null,
+				post_id: postId,
+				post_reactions: getDefaultReactionObj(),
+				proposalHashBlock: proposalHashBlock || null,
+				proposer: proposer || subsquidPost?.preimage?.proposer || otherPostProposer || curator || null,
+				requestedAmount: requestedAmt ? requestedAmt.toString() : undefined,
+				status: status,
+				status_history: statusHistory,
+				tally,
+				timeline: proposalTimeline,
+				title: subsquareTitle || 'Untitled',
+				topic: topicFromType,
+				track_no: !isNaN(trackNumber) ? trackNumber : null,
+				type: type,
+				user_id: 1
+			};
+			await getContentSummary(post, network, true);
+
+			if (!post?.content?.length) {
+				if (proposer) {
+					post.content = `This is a ${getProposalTypeTitle(
```

Comment:
please make this text into a util function, if we do not have one used in FE already